crates/testlang/outputs/npm/tests/src/tests/query.ts

```diff
@@ -0,0 +1,58 @@
+import { TokenNode } from "@slang-private/slang-testlang/cst";
+import { RuleKind } from "@slang-private/slang-testlang/kinds";
+import { Language } from "@slang-private/slang-testlang/language";
+import { Query } from "@slang-private/slang-testlang/query";
+
+test("simple query", () => {
+  const language = new Language("1.0.0");
+  const tree_source = `tree [A [B C] D];`;
+  const parse_output = language.parse(RuleKind.Tree, tree_source);
+
+  const query_source = `[TreeNodeChild ... @id [DelimitedIdentifier]]`;
+  const query = Query.parse(query_source);
+
+  const query_result = parse_output.createTreeCursor().query([query]);
+  {
+    let result = query_result.next();
+    let cursors = result!.bindings["id"]!;
+    expect(cursors.length).toEqual(1);
+    expect((cursors[0]!.node() as TokenNode).text).toEqual("A");
```

Comment:
I suggest refactoring into a common local util, and calling for each result:

```ts
const expectNextResult = (name) => {
  // ...
};

expectNextResult("A");
expectNextResult("B");
// ....
```

Reply:
Done